src/index.ts

```diff
@@ -180,411 +77,142 @@ export function resolve(
     }
   }
 
-  initMappers(cachedOptions)
+  if (!resolver) {
+    const optionsHash = stableHash(options)
+    options = normalizeOptions(options)
+    // take `cwd` into account -- #217
+    const cacheKey = `${optionsHash}:${process.cwd()}`
+    let cached = resolverCache.get(cacheKey)
+    if (!cached && !options.project) {
+      resolverCache.set(cacheKey, (cached = new ResolverFactory(options)))
+    }
+    resolver = cached
+  }
 
-  let mappedPaths = getMappedPaths(source, file, cachedOptions.extensions, true)
+  source = removeQuerystring(source)
 
-  if (mappedPaths.length > 0) {
-    log('matched ts path:', ...mappedPaths)
-  } else {
-    mappedPaths = [source]
-  }
+  options ||= {}
+
+  // eslint-disable-next-line sonar/label-position, sonar/no-labels
+  resolve: if (!resolver) {
```

Comment:
Maybe we should change the name of this label? There are way too many `resolve`s here. 

Maybe `create_resolver` or something?